tests/cuda_ndarrays/test_cuda_ndarrays.cu

```diff
@@ -0,0 +1,537 @@
+        const char * func, const char *file, int32_t line)
+{
+    if (v1 != v2)
+    {
+        printf("[FAIL] %s:%d:%s\n", file, line, func);
+        printf("[INFO] %s:%f != %s:%f\n", getname(v1), v1, getname(v2),v2);
+        printf("[DSCR] %s\n", dscr);
+        return ;
+    }
+    printf("[PASS] %s:%d:%s\n", file, line, func);
+    printf("[DSCR] %s\n", dscr);
+}
+
+void assert_int64(int64_t v1, int64_t v2, const char *dscr,
+        const char * func, const char *file, int32_t line)
+{
+    if (v1 != v2)
+    {
+        printf("[FAIL] %s:%d:%s\n", file, line, func);
+        printf("[INFO] %s:%ld != %s:%ld\n", getname(v1), v1, getname(v2),v2);
+        printf("[DSCR] %s\n", dscr);
+        return ;
+    }
+    printf("[PASS] %s:%d:%s\n", file, line, func);
+    printf("[DSCR] %s\n", dscr);
+}
+
+void assert_int32(int32_t v1 , int32_t v2, const char *dscr,
+        const char * func, const char *file, int32_t line)
+{
+    if (v1 != v2)
+    {
+        printf("[FAIL] %s:%d:%s\n", file, line, func);
+        printf("[INFO] %s:%d != %s:%d\n", getname(v1), v1, getname(v2),v2);
+        printf("[DSCR] %s\n", dscr);
+        return ;
+    }
+    printf("[PASS] %s:%d:%s\n",file, line, func);
+    printf("[DSCR] %s\n", dscr);
+}
+
+void assert_int16(int16_t v1 , int16_t v2, const char *dscr,
+        const char * func, const char *file, int32_t line)
+{
+    if (v1 != v2)
+    {
+        printf("[FAIL] %s:%d:%s\n", file, line, func);
+        printf("[INFO] %s:%d != %s:%d\n", getname(v1), v1, getname(v2),v2);
+        printf("[DSCR] %s\n", dscr);
+        return ;
+    }
+    printf("[PASS] %s:%d:%s\n", file, line, func);
+    printf("[DSCR] %s\n", dscr);
+}
+
+void assert_int8(int8_t v1 , int8_t v2, const char *dscr,
+        const char * func, const char *file, int32_t line)
+{
+    if (v1 != v2)
+    {
+        printf("[FAIL] %s:%d:%s\n", file, line, func);
+        printf("[INFO] %s:%d != %s:%d\n", getname(v1), v1, getname(v2),v2);
+        printf("[DSCR] %s\n", dscr);
+        return ;
+    }
+    printf("[PASS] %s:%d:%s\n", file, line, func);
+    printf("[DSCR] %s\n", dscr);
+}
+
+void test_cuda_array_create_host_double()
+{
+    t_ndarray arr = {.shape = NULL};
+
+    int64_t tmp_shape[] = {INT64_C(14)};
+
+    arr = cuda_array_create(1, tmp_shape, nd_double, false, allocateMemoryOnHost);
+    double cuda_array_dummy[] = {1.02, 0.25, 5e-05, 1.0, 200.0, 33.0, 5.0, 57.0, 62.0, 70.0, 103.009, 141.0, 122.0, 26.5};
+    cudaMemcpy(arr.nd_double, cuda_array_dummy, arr.buffer_size, cudaMemcpyHostToHost);
+
+    assert_int32(arr.nd, 1, "testing the number of dimensions", __func__, __FILE__, __LINE__);
+    assert_int64(arr.shape[0], tmp_shape[0], "testing the shape", __func__, __FILE__, __LINE__);
+    for(int i = 0; i < tmp_shape[0]; i++)
+    {
+        assert_double(arr.nd_double[i], cuda_array_dummy[i], "testing the data", __func__, __FILE__, __LINE__);
+    }
+    cuda_free_host(arr);
+}
+
+void test_cuda_array_create_managed_double()
```

Comment:
why should we create another array to check the `managed_memory` array

Reply:
Actually, there is no need to create an extra array if I only needed to access the data. I just forgot to update it.
My original intention was to have a separate copy of the data in host memory and apply some operations on it before applying the same operations on the original array from a kernel. What do you think should I proceed , or keeping it simple for this specific test may be sufficient ?

Reply:
I think just doing the same as the test for `host_memory` without using the additional array.
the other tests will be added when we add `kernel` tests.